client/client.go

```diff
@@ -418,17 +419,59 @@ func (r *NotaryRepository) ListTargets() ([]*Target, error) {
 		return nil, err
 	}
 
+	if len(roles) == 0 {
+		roles = []string{data.CanonicalTargetsRole}
+	}
+	targets := make(map[string]*Target)
+	for _, role := range roles {
+		// we don't need to do anything special with removing role from
+		// roles because listSubtree always processes role and only excludes
+		// descendent delegations that appear in roles.
+		r.listSubtree(targets, role, roles...)
```

Comment:
Just want to verify this behavior -  if I have (an admittedly stupid) priority list like: `"targets/a", "targets/x", "targets/a/b", this would list *only* the parts of `targets/a` that are not in `targets/a/b` first.  So that targets in `targets/x` would have priority over in targets in `targets/a/b` even though `targets/a` is listed first.

Is that correct?  Or doe the whole `targets/a` subtree come first?  If so, we should probably exclude only the previous roles.

I'm fine with this either way, and the current way makes sense, but what happens if a subtree of a higher priority role is given lower priority than some other role is kind of confusing.  It might be good to add a test for it and clearer docstrings.